src/execution.rs

```diff
@@ -484,6 +492,23 @@ impl Conditional {
     }
 }
 
+impl ForIn {
+    fn execute(&self, exec: &mut ExecutionContext) -> Result<(), ExecutionError> {
+        let values = self.values.evaluate(exec)?;
+        let values = values.into_list(exec.graph)?;
+        for value in values {
+            exec.locals.add_immutable(self.name, value).map_err(|_| {
+                ExecutionError::DuplicateVariable(format!("{}", self.name.display_with(exec.ctx)))
+            })?;
+            for stmt in &self.statements {
+                stmt.execute(exec)?;
+            }
+            exec.locals.remove(self.name);
```

Comment:
I am happy to see that the loop variable doesn't persist after the loop is done. :+1:

Reply:
:)